x-pack/legacy/plugins/siem/public/lib/connectors/servicenow.tsx

```diff
@@ -0,0 +1,222 @@
+      }
+
+      if (!action.secrets.username) {
+        errors.username = [...errors.username, i18n.SERVICENOW_USERNAME_REQUIRED];
+      }
+
+      if (!action.secrets.password) {
+        errors.password = [...errors.password, i18n.SERVICENOW_PASSWORD_REQUIRED];
+      }
+
+      return { errors };
+    },
+    validateParams: (actionParams: ServiceNowActionParams): ValidationResult => {
+      const errors = {
+        message: [] as string[],
+      };
+      return { errors };
+    },
+    actionConnectorFields: ServiceNowConnectorFields,
+    actionParamsFields: ServiceNowParamsFields,
+  };
+}
+
+const ServiceNowConnectorFields: React.FunctionComponent<ActionConnectorFieldsProps<
+  ServiceNowActionConnector
+>> = ({ action, editActionConfig, editActionSecrets, errors }) => {
+  const { apiUrl, casesConfiguration: { mapping = [] } = {} } = action.config;
+  const { username, password } = action.secrets;
+
+  const isApiUrlInvalid: boolean = errors.apiUrl.length > 0 && apiUrl != null;
+  const isUsernameInvalid: boolean = errors.username.length > 0 && username != null;
+  const isPasswordInvalid: boolean = errors.password.length > 0 && password != null;
+
+  if (isEmpty(mapping)) {
+    editActionConfig('casesConfiguration', {
+      ...action.config.casesConfiguration,
+      mapping: defaultMapping,
+    });
+  }
+
+  const handleOnChangeActionConfig = useCallback(
+    (key: string, evt: ChangeEvent<HTMLInputElement>) => editActionConfig(key, evt.target.value),
+    []
+  );
+
+  const handleOnBlurActionConfig = useCallback(
+    (key: string) => {
+      if (key === 'apiUrl' && action.config[key] == null) {
+        editActionConfig(key, '');
+      }
+    },
+    [action.config]
+  );
+
+  const handleOnChangeSecretConfig = useCallback(
+    (key: string, evt: ChangeEvent<HTMLInputElement>) => editActionSecrets(key, evt.target.value),
+    []
+  );
+
+  const handleOnBlurSecretConfig = useCallback(
+    (key: string) => {
+      if (['username', 'password'].includes(key) && get(key, action.secrets) == null) {
+        editActionSecrets(key, '');
+      }
+    },
+    [action.secrets]
+  );
+
+  const handleOnChangeMappingConfig = useCallback(
+    (newMapping: CasesConfigurationMapping[]) =>
+      editActionConfig('casesConfiguration', {
+        ...action.config.casesConfiguration,
+        mapping: newMapping,
+      }),
+    [action.config]
+  );
+
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="apiUrl"
+            fullWidth
+            error={errors.apiUrl}
+            isInvalid={isApiUrlInvalid}
+            label={i18n.SERVICENOW_API_URL_LABEL}
+          >
+            <EuiFieldText
+              fullWidth
+              isInvalid={isApiUrlInvalid}
+              name="apiUrl"
+              value={apiUrl}
+              data-test-subj="apiUrlFromInput"
+              placeholder="https://<instance>.service-now.com"
```

Comment:
nit: `service-now` to `servicenow` to follow their website

Reply:
Instances URLs have an `-` between `service` and `now`. The official site does not.

Reply:
that's a good point we should have a config in siem kibana for it just in case. what do you think?